unitTests/ui/PrintableUriTest.ts

```diff
@@ -47,22 +60,65 @@ export function PrintableUriTest() {
       fakeResult = null;
     });
 
-    it('should display the XML correctly if the result has a non-null parents field', () => {
-      fakeResult.raw.parents =
-        '<?xml version="1.0" encoding="utf-16"?><parents><parent name="My Drive" uri="https://drive.google.com/#my-drive" /></parents>';
-      test = Mock.advancedResultComponentSetup<PrintableUri>(PrintableUri, fakeResult, undefined);
-      expect($$(test.cmp.element).find('.CoveoResultLink').innerText).toEqual('My Drive');
+    describe('with a non-null parents field', () => {
+      beforeEach(() => {
+        fakeResult.raw.parents =
+          '<?xml version="1.0" encoding="utf-16"?><parents><parent name="My Drive" uri="https://drive.google.com/#my-drive" /></parents>';
+        test = Mock.advancedResultComponentSetup<PrintableUri>(PrintableUri, fakeResult, undefined);
+      });
+
+      it('should display the XML correctly', () => {
+        expect($$(test.cmp.element).find('.CoveoResultLink').innerText).toEqual('My Drive');
+      });
+
+      it('should not display an ellipsis', () => {
+        expect(getEllipsis().length).toEqual(0);
+      });
     });
 
-    it('should display the XML correctly if the result has a very long parents field', () => {
-      fakeResult.raw.parents =
-        '<?xml version="1.0" encoding="utf-16"?><parents><parent name="Organization" uri="https://na17.salesforce.com/home/home.jsp" />`' +
-        '<parent name="Technical_Article__ka" uri="http://www.salesforce.com/org:organization/articletype:Technical_Article" />`' +
-        '<parent name="Generator became sentient and refuses to cut power"`' +
-        ' uri="https://na17.salesforce.com/kA0o00000003Wpk" /><parent name="un autre test" uri="http//:google.ca" /></parents>';
-      test = Mock.advancedResultComponentSetup<PrintableUri>(PrintableUri, fakeResult, undefined);
-      expect($$(test.cmp.element).findAll('.coveo-printable-uri-part')[0].innerText).toEqual('Organization');
-      expect($$(test.cmp.element).findAll('.coveo-printable-uri-part')[1].innerText).toEqual('Technical_Article__ka');
+    describe('with a very long parents field', () => {
+      beforeEach(() => {
+        fakeResult.raw.parents = longParentsXml;
+        test = Mock.advancedResultComponentSetup<PrintableUri>(PrintableUri, fakeResult, undefined);
+      });
+
+      it('should display the first 4 parents and the last parent', () => {
+        expect(getURIPartsNames()).toEqual([...range(4), longParentsCount - 1].map(index => index.toString()));
+      });
+
+      it('should display an ellipsis', () => {
+        expect(getEllipsis().length).toEqual(1);
+      });
+
+      describe('after clicking on the ellipsis', () => {
+        beforeEach(() => {
+          const [ellipsis] = getEllipsis();
+          (ellipsis.firstChild as HTMLElement).click();
+        });
+
+        it('should display the next 4 parents and the last parent', () => {
```

Comment:
Those tests will work for desktop, is it unit testing the mobile settings?
`partsCount = DeviceUtils.isMobileDevice() ? 3 : 5`

Reply:
I'll try to make it test mobile settings.